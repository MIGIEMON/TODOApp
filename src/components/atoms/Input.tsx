import './Input.css';

type InputProps = {
  inputType: string;
  value: string;
  onChange: (value: string) => void;
  placeholder?: string;
  type?: string;
};

export const Input: React.FC<InputProps> = ({ inputType, value, onChange, placeholder, type = 'text' }) => {
  return (
    <input
      className={`Input ${inputType}`}
      type={type}
      value={value}
      onChange={(e) => onChange(e.target.value)}
      placeholder={placeholder}
    />
  );
};