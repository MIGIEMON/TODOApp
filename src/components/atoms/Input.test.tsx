import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Input } from './Input';

describe('Inputコンポーネント', () => {
  const handleChange = jest.fn();

  it('インプットにvalueが入った状態で表示される', () => {
    const { container } = render(
      <Input inputType="test" value="テスト" onChange={handleChange} />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('valueを変更するとonChangeが呼び出される', () => {
    const { getByDisplayValue } = render(
      <Input inputType="test" value="テスト" onChange={handleChange} />
    );
    fireEvent.change(getByDisplayValue('テスト'), { target: { value: 'テスト2' } });
    expect(handleChange).toHaveBeenCalledTimes(1);
    expect(handleChange).toHaveBeenCalledWith('テスト2');
  });

  it('プレースホルダーが表示される', () => {
    const { getByPlaceholderText } = render(
      <Input
        inputType="test"
        value=""
        onChange={handleChange}
        placeholder="プレースホルダー"
      />
    );
    expect(getByPlaceholderText('プレースホルダー')).toBeInTheDocument();
  });
});
