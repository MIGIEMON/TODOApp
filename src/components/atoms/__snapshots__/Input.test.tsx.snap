// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inputコンポーネント インプットにvalueが入った状態で表示される 1`] = `
<input
  class="Input test"
  type="text"
  value="テスト"
/>
`;
